Source/JavaWrappers.cpp

```diff
@@ -274,6 +282,112 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    jclass g_webSocketClass{};
+    void InitializeJavaWebSocketClass(jclass webSocketClass, JNIEnv* env) {
+        g_webSocketClass = (jclass) env->NewGlobalRef(webSocketClass);
+    }
+    void DestructJavaWebSocketClass(JNIEnv* env) {
+        env->DeleteGlobalRef(g_webSocketClass);
+    }
+    std::vector<std::pair<jobject, WebSocketClient*>> WebSocketClient::s_objectVector;
+
+    WebSocketClient::WebSocketClient(std::string url, std::function<void()> open_callback, std::function<void()> close_callback, std::function<void(std::string)> message_callback, std::function<void()> error_callback)
+        : Object{g_webSocketClass}
+        , m_openCallback{std::move(open_callback)}
+        , m_messageCallback{std::move(message_callback)}
+        , m_closeCallback{std::move(close_callback)}
+        , m_errorCallback{std::move(error_callback)}
+    {
+        static JNINativeMethod methods[] = {
+                {"closeCallback", "()V", (void*)OnClose },
+                {"openCallback", "()V", (void*)&WebSocketClient::OnOpen },
+                {"messageCallback", "(Ljava/lang/String;)V", (void*)&WebSocketClient::OnMessage },
+                {"errorCallback", "()V", (void*)&WebSocketClient::OnError },
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        JObject(m_env->NewObject(m_class, m_env->GetMethodID(m_class, "<init>", "(Ljava/lang/String;)V"), m_env->NewStringUTF(url.c_str())));
+
+        s_objectVector.push_back(std::make_pair(JObject(), this));
+
+        jmethodID connectSocket{m_env->GetMethodID(m_class, "connectBlocking", "()Z")};
+        m_env->CallBooleanMethod(JObject(), connectSocket);
+    }
+
+    WebSocketClient::~WebSocketClient()
+    {
+        WebSocketClient::s_objectVector.erase(std::remove_if(WebSocketClient::s_objectVector.begin(), WebSocketClient::s_objectVector.end(), [this](const auto& p) { return p.second == this; }), WebSocketClient::s_objectVector.end());
```

Comment:
I don't understand why there is an `erase` call after getting the removed iterator. Also, it's erasing to the end? What does this do? :-)

```suggestion
        std::remove_if(WebSocketClient::s_objectVector.begin(), WebSocketClient::s_objectVector.end(), [this](const auto& p) { return p.second == this; });
```

Reply:
This is what I was following: https://en.cppreference.com/w/cpp/algorithm/remove#:~:text=A%20call%20to,standard%20sequence%20containers%2C. Can follow up offline.

Reply:
Spoke offline, changed to for loop.